platform/security/pdp/security-pdp-xacmlrealm/src/main/java/ddf/security/pdp/xacml/processor/BalanaPdp.java

```diff
@@ -330,28 +318,29 @@ public void endElement(String uri, String localName, String qName)
     /**
      * Marshalls the XACML request to a string.
      *
-     * @param xacmlRequestType
-     *            The XACML request to marshal.
+     * @param xacmlRequestType The XACML request to marshal.
      * @return A string representation of the XACML request.
      */
     private String marshal(RequestType xacmlRequestType) throws PdpException {
-        ObjectFactory objectFactory = new ObjectFactory();
-        Writer writer = new StringWriter();
-        JAXBElement<RequestType> xacmlRequestTypeElement = objectFactory
-                .createRequest(xacmlRequestType);
-
-        Marshaller marshaller = null;
         String xacmlRequest = null;
-
+        if (null == parser) {
+            throw new IllegalStateException("XMLParser must be configured.");
+        }
         try {
-            marshaller = jaxbContext.createMarshaller();
-            marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
-            marshaller.marshal(xacmlRequestTypeElement, writer);
-
-            xacmlRequest = writer.toString();
-        } catch (JAXBException e) {
+            List<String> ctxPath = new ArrayList<>(2);
+            ctxPath.add(RequestType.class.getPackage()
+                    .getName());
+            ctxPath.add(ResponseType.class.getPackage()
+                    .getName());
+            ParserConfigurator configurator = parser.configureParser(ctxPath,
```

Comment:
This configurator needs the `Marshaller.JAXB_FORMATTED_OUTPUT` property set to true to match old behavior.